frontend/packages/dev-console/src/components/topology/data-transforms/__tests__/data-transformer.spec.ts

```diff
@@ -212,20 +212,33 @@ describe('data transformer ', () => {
     const mockCheURL = 'https://mock-che.test-cluster.com';
     const mockGitURL =
       mockResources.deploymentConfigs.data[0].metadata.annotations['app.openshift.io/vcs-uri'];
+    const mockGitBranch =
+      mockResources.deploymentConfigs.data[0].metadata.annotations['app.openshift.io/vcs-ref'];
     const graphData = getTransformedTopologyData(mockResources, ['deploymentConfigs']);
-    const generatedEditURL = getEditURL(mockGitURL, mockCheURL);
-    const { editURL, vcsURI } = graphData.nodes[0].data.data as WorkloadData;
-    const editUrl = editURL || getEditURL(vcsURI, mockCheURL);
+    const generatedEditURL = getEditURL(mockGitURL, mockGitBranch, mockCheURL);
+    const { editURL, vcsURI, vcsRef } = graphData.nodes[0].data.data as WorkloadData;
+    const editUrl = editURL || getEditURL(vcsURI, vcsRef, mockCheURL);
 
     expect(editUrl).toBe(generatedEditURL);
   });
 
   it('should return the git repo URL if cheURL is not there', () => {
+    const mockGitURL =
+      mockResources.deploymentConfigs.data[0].metadata.annotations['app.openshift.io/vcs-uri'];
+    const mockGitBranch =
+      mockResources.deploymentConfigs.data[0].metadata.annotations['app.openshift.io/vcs-ref'];
+    const graphData = getTransformedTopologyData(mockResources, ['deploymentConfigs']);
+    const { editURL, vcsURI, vcsRef } = graphData.nodes[0].data.data as WorkloadData;
+    const editUrl = editURL || getEditURL(vcsURI, vcsRef, '');
+    expect(editUrl).toBe(`${mockGitURL}/tree/${mockGitBranch}`);
+  });
+
+  it('should return only the git repo URL if branch name is not provided', () => {
     const mockGitURL =
       mockResources.deploymentConfigs.data[0].metadata.annotations['app.openshift.io/vcs-uri'];
     const graphData = getTransformedTopologyData(mockResources, ['deploymentConfigs']);
     const { editURL, vcsURI } = graphData.nodes[0].data.data as WorkloadData;
-    const editUrl = editURL || getEditURL(vcsURI, '');
+    const editUrl = editURL || getEditURL(vcsURI, '', '');
```

Comment:
```suggestion
    const editUrl = getEditURL(vcsURI, '', '');
```

Please remove this "logic" from the test. Each test case should test a strict behavior and I expect that such a fallback is what we want "to test" here.

Reply:
Removed.